### PR TITLE
Reorganized documentation and added new articles

### PR DESCRIPTION
**Pull request by mdurrani808**
Added:
* networking diagram
* 3D Coordiante systems
* Reorganized getting started
* Removed AprilTag example page
* Added note on AruCo vs AprilTag
* Added page on MultiTag, needs docs
* Basic system requirements

Did some basic reorganization as well to make this more approachable.

### CODE REVIEW

**Comment by mdurrani808** [comment]
no clue why the Chief Delphi links are failing now

File: source/docs/apriltag-pipelines/3D-tracking.rst [hunk 1]
```diff
@@ -0,0 +1,16 @@
+3D Tracking
+===========
+
+3D AprilTag tracking will allow you to track XYZ position and orientation of a tag in the camera frame. This is useful for robot pose estimation and other applications like autonomous scoring. In order to use 3D tracking, you must first :ref:`calibrate your camera <docs/calibration/calibration:Calibrating Your Camera>`. Once you have, you need to enable 3D mode in the UI and you will now be able to get 3D pose information from the tag! For information on getting and using this information in your code, see :ref:`the programming reference. <docs/programming/index:Programming Reference>`.
```
Comment (gerth2) [inline source/docs/apriltag-pipelines/3D-tracking.rst:4]: XYZ Position and orientation seems ambigous to me....

Comment (gerth2) [reply source/docs/apriltag-pipelines/3D-tracking.rst:4]: bah. recommendation...

"3D AprilTag tracking will allow you to track the real-world position and rotation of a tag relative to the camera's image sensor."

File: source/docs/apriltag-pipelines/3D-tracking.rst [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+Ambiguity
+---------
+Translating from 2D to 3D using data from the calibration and the four tag corners can lead to "pose ambiguity", where it appears that the AprilTag pose is flipping between two different poses. You can read more about this issue `here. <https://docs.wpilib.org/en/stable/docs/software/vision-processing/apriltag/apriltag-intro.html#d-to-3d-ambiguity>`
+
+VIDEO HERE
```
Comment (gerth2) [inline source/docs/apriltag-pipelines/3D-tracking.rst:10]:
```suggestion
Video coming soon!
```

File: source/docs/apriltag-pipelines/3D-tracking.rst (hunk 1 above)
Comment (gerth2) [inline source/docs/apriltag-pipelines/3D-tracking.rst:4]:
```suggestion
3D AprilTag tracking will allow you to track the real-world position and rotation of a tag relative to the camera's image sensor. This is useful for robot pose estimation and other applications like autonomous scoring. In order to use 3D tracking, you must first :ref:`calibrate your camera <docs/calibration/calibration:Calibrating Your Camera>`. Once you have, you need to enable 3D mode in the UI and you will now be able to get 3D pose information from the tag! For information on getting and using this information in your code, see :ref:`the programming reference. <docs/programming/index:Programming Reference>`.
```

File: source/docs/apriltag-pipelines/about-apriltags.rst [hunk 3]
```diff
@@ -0,0 +1,13 @@
+About Apriltags
+===============
+
+.. image:: images/pv-apriltag.png
+   :align: center
+   :scale: 20 %
+
+AprilTags are a type of visual fiducial marker that is commonly used within robotics and computer vision applications. Visual fiducial markers are artificial landmarks added to a scene to allow "localization" (finding your current position) via images. In simpler terms, it is something that can act as a known point of reference that you can use to find your current location. They are similar to QR codes in which they encode information, however, they hold much less data. This has the added benefit of being much easier to track from long distances and at low resolutions. By placing AprilTags in known locations around the field and detecting them using PhotonVision, you can easily get full field localization / pose estimation. Alternatively, you can use AprilTags the same way you used retroreflective tape, simply using them to turn to goal without any pose estimation.
```
Comment (gerth2) [inline source/docs/apriltag-pipelines/about-apriltags.rst:8]:
```suggestion
AprilTags are a common type of visual fiducial marker. Visual fiducial markers are artificial landmarks added to a scene to allow "localization" (finding your current position) via images. In simpler terms, tags mark known points of reference that you can use to find your current location. They are similar to QR codes in which they encode information, however, they hold only a single number. By placing AprilTags in known locations around the field and detecting them using PhotonVision, you can easily get full field localization / pose estimation. Alternatively, you can use AprilTags the same way you used retroreflective tape, simply using them to turn to goal without any pose estimation.
```

Comment (gerth2) [reply source/docs/apriltag-pipelines/about-apriltags.rst:8]: The removed sentence is a good point, but wasn't relevant to the rest of the paragraph flow, so I popped it out.

File: source/docs/apriltag-pipelines/coordinate-systems.rst [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+You define the camera to robot transform in the robot coordinate frame.
+
+Camera Coordinate Frame
+-----------------------
+
+The camera coordinate system is defined as follows, relative to the camera sensor itself:
+
+* The origin is the center.
+* The x-axis points to the left (when looking at the camera sensor from the front)
```
Comment (gerth2) [inline source/docs/apriltag-pipelines/coordinate-systems.rst:17]:
```suggestion
* The x-axis points to the left
```

File: source/docs/apriltag-pipelines/coordinate-systems.rst [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+--------------------------------
+
+PhotonVision follows the WPILib conventions for the robot and field coordinate-systems, as defined `here <https://docs.wpilib.org/en/stable/docs/software/advanced-controls/geometry/coordinate-systems.html>`_.
+
+You define the camera to robot transform in the robot coordinate frame.
+
+Camera Coordinate Frame
+-----------------------
+
+The camera coordinate system is defined as follows, relative to the camera sensor itself:
```
Comment (gerth2) [inline source/docs/apriltag-pipelines/coordinate-systems.rst:14]:
```suggestion
The camera coordinate system is defined as follows, relative to the camera sensor itself, and when looking in the same direction as the sensor points:
```

File: source/docs/apriltag-pipelines/coordinate-systems.rst [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+* The origin is the center.
+* The x-axis points to the left (when looking at the camera sensor from the front)
+* The y-axis points up.
+* The z-axis points out.
```
Comment (gerth2) [inline source/docs/apriltag-pipelines/coordinate-systems.rst:19]:
```suggestion
* The z-axis points out toward the subject.
```

File: source/docs/apriltag-pipelines/coordinate-systems.rst [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+AprilTag Coordinate Frame
+-------------------------
+
+The AprilTag coordinate system is defined as follows, relative to the center of the AprilTag itself:
```
Comment (gerth2) [inline source/docs/apriltag-pipelines/coordinate-systems.rst:24]:
```suggestion
The AprilTag coordinate system is defined as follows, relative to the center of the AprilTag itself, and when viewing the tag as a robot would.=:
```

File: source/docs/apriltag-pipelines/coordinate-systems.rst [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+* The origin is the center.
+* The x-axis points to the right when looking at the tag straight on.
```
Comment (gerth2) [inline source/docs/apriltag-pipelines/coordinate-systems.rst:27]:
```suggestion
* The x-axis points to your right
```

File: source/docs/apriltag-pipelines/coordinate-systems.rst [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+* The y-axis points upwards.
+
```
Comment (gerth2) [inline source/docs/apriltag-pipelines/coordinate-systems.rst:29]:
```suggestion
 * The z-axis is normal to the plane the tag is printed on, pointing outward from the visible side of the tag.
```

File: source/docs/apriltag-pipelines/images/camera-coord.png
Comment (gerth2) [inline source/docs/apriltag-pipelines/images/camera-coord.png:1]: The description says Z points out the axis of the camera, I think either this picture or the description is wrong?

File: source/docs/hardware/selecting-hardware.rst [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+
+Reccomended Cameras
+^^^^^^^^^^^^^^^^^^^
+For colored shape detection, any non-fisheye camera supported by PhotonVision will work. We reccomend the Pi Camera V1 or a high fps USB camera. 
+
+For driver camera, we reccomend a USB camera with a fisheye lens, so your driver can see more of the field. 
+
+For AprilTag detection, we reccomend you use a global shutter camera that has ~100 degree diagonal FOV. This will allow you to see more AprilTags in frame, and will allow for more accurate pose estimation. You also want a camera that supports high FPS, as this will allow you to update your pose estimator at a higher frequency. 
+
+* Reccomendations For AprilTag Detection
```
Comment (gerth2) [inline source/docs/hardware/selecting-hardware.rst:57]: ... photonvision amazon affiliate links time?

File: source/docs/programming/photonlib/robot-pose-estimator.rst [hunk 11]
```diff
@@ -33,6 +33,9 @@ Creating a ``PhotonPoseEstimator``
 ----------------------------------
 The PhotonPoseEstimator has a constructor that takes an ``AprilTagFieldLayout`` (see above), ``PoseStrategy``, ``PhotonCamera``, and ``Transform3d``. ``PoseStrategy`` has six possible values:
 
+* MULTI_TAG_PNP_ON_COPROCESSOR
+    *  Calculates a new robot position estimate by combining all visible tag corners. Reccomended for all teams as it will be the most accurate.
```
Comment (gerth2) [inline source/docs/programming/photonlib/robot-pose-estimator.rst:37]:
```suggestion
    *  Calculates a new robot position estimate by combining all visible tag corners. Recommended for all teams as it will be the most accurate.
```

**Review by gerth2 — DISMISSED** [review]
very solid overall. A few wording nits

File: source/docs/hardware/selecting-hardware.rst (hunk 10 above)
Comment (mdurrani808) [reply source/docs/hardware/selecting-hardware.rst:57]: Link check doesn't like amazon links so unless I can figure out how to exclude those..

File: source/docs/apriltag-pipelines/3D-tracking.rst [hunk 12]
```diff
@@ -0,0 +1,16 @@
+3D Tracking
+===========
+
+3D AprilTag tracking will allow you to track the real-world position and rotation of a tag relative to the camera's image sensor. This is useful for robot pose estimation and other applications like autonomous scoring. In order to use 3D tracking, you must first :ref:`calibrate your camera <docs/calibration/calibration:Calibrating Your Camera>`. Once you have, you need to enable 3D mode in the UI and you will now be able to get 3D pose information from the tag! For information on getting and using this information in your code, see :ref:`the programming reference. <docs/programming/index:Programming Reference>`.
+
+Ambiguity
+---------
+Translating from 2D to 3D using data from the calibration and the four tag corners can lead to "pose ambiguity", where it appears that the AprilTag pose is flipping between two different poses. You can read more about this issue `here. <https://docs.wpilib.org/en/stable/docs/software/vision-processing/apriltag/apriltag-intro.html#d-to-3d-ambiguity>`
```
Comment (mcm001) [inline source/docs/apriltag-pipelines/3D-tracking.rst:8]: We should define how ambiguity is calculated

File: source/docs/apriltag-pipelines/3D-tracking.rst [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+
+VIDEO HERE
```
Comment (mcm001) [inline source/docs/apriltag-pipelines/3D-tracking.rst:10]: Video here

File: source/docs/contributing/photonvision/build-instructions.rst [hunk 14]
```diff
@@ -93,6 +93,32 @@ Running the following command under the root directory will build the jar under
 
       ``gradlew shadowJar``
 
+Build and Run the Source on a Raspberry Pi Coprocessor
```
Comment (mcm001) [inline source/docs/contributing/photonvision/build-instructions.rst:96]: "The source"?

File: source/docs/contributing/photonvision/build-instructions.rst [hunk 15]
```diff
@@ -165,6 +191,39 @@ One way is by running the program using gradle with the :code:`--debug-jvm` flag
...
+            "hostName": "photonvision.local",
+            "port": "5801",
+            "projectName": "photon-core"
+        },
+      ]
+   }
+
+Stop any existing instance of PhotonVision.
+
+Launch the program with the following additional argument to the JVM: :code:`java -agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=photonvision.local:5801`
```
Comment (mcm001) [inline source/docs/contributing/photonvision/build-instructions.rst:221]: The address= block seems fishy to me

File: source/docs/pipelines/input.rst [hunk 16]
```diff
@@ -21,18 +21,10 @@ Camera exposure and brightness control how bright the captured image will be, al
...
-^^^^^^^^^^^^^^^^^^^^^^^^^
-For AprilTag pipelines, your goal is to reduce the "motion blur" as much as possible. Motion blur is the visual streaking/smearing on the camera stream as a result of movement of the camera or object of focus. You want to mitigate this as much as possible because your robot is constantly moving and you want to be able to read as many tags as you possibly can. The possible solutions to this include:
 
-1. Cranking your exposure as low as it goes and increasing your gain/brightness. This will decrease the effects of motion blur and increase FPS.
-2. Using a global shutter (as opposed to rolling shutter) camera. This should eliminate most, if not all motion blur.
-3. Only rely on tags when not moving.
-
-.. image:: images/motionblur.gif
-    :align: center
+For AprilTag/AruCo pipelines, you want to reduce the "motion blur" as much as possible by lowing your exposure and increasing brightness. Motion blur is the visual streaking/smearing on the camera stream as a result of movement of the camera or object of focus.
```
Comment (mcm001) [inline source/docs/pipelines/input.rst:27]: Why delete this section on exposure?

File: source/docs/contributing/photonvision/build-instructions.rst (hunk 15 above)
Comment (mdurrani808) [reply source/docs/contributing/photonvision/build-instructions.rst:221]: @gerth2 ^ thoughts on above

Comment (mcm001) [reply source/docs/contributing/photonvision/build-instructions.rst:221]: From https://www.jetbrains.com/help/idea/tutorial-remote-debug.html#run-app:

java -jar -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005 remote-debug.jar

File: source/docs/apriltag-pipelines/3D-tracking.rst [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+
+Ambiguity
+---------
+
+Translating from 2D to 3D using data from the calibration and the four tag corners can lead to "pose ambiguity", where it appears that the AprilTag pose is flipping between two different poses. You can read more about this issue `here. <https://docs.wpilib.org/en/stable/docs/software/vision-processing/apriltag/apriltag-intro.html#d-to-3d-ambiguity>` Ambiguity is calculated as the ratio of reprojection errors between two pose solutions (if they exist), where reprojection error is the error corresponding to the image distance between a projected point and measured one.
```
Comment (mcm001) [inline source/docs/apriltag-pipelines/3D-tracking.rst:9]: Can we be more specific than "a measured pose? I would suggest "between where the apriltag's corners are detected vs where we expect to see them based on the tag's estimated camera relative pose"

**Review by mcm001 — APPROVED** [review]

File: source/docs/apriltag-pipelines/3D-tracking.rst (hunk 13 above)
Comment (mdurrani808) [reply source/docs/apriltag-pipelines/3D-tracking.rst]: punted
